### DISCUSSION TITLE
connectGeoSearch with multiple locations

### DISCUSSION THREAD

**Discussion by samouss** [algolia/instantsearch#5211] [Ideas]
**Do you want to request a *feature* or report a *bug*?**

Feature

**Feature: What is your use case for such a feature?**

In some cases hits can have multiple locations. In that case the `_geoloc` attribute will be an array of position instead of being a single location object. Right now we only handle the case for single location.

**Feature: What is your proposed API entry? The new option to add? What is the behavior?**

There are several things to do in order to support this behaviour:

 - enable `getRankingInfo` on the `SearchParameters` in order to have access to the `_rankingInfo` attribute that contains [the correct matched location](https://www.algolia.com/doc/guides/searching/geo-search/?language=javascript#identifying-the-matching-geo-search-with-rankinginfo)
- **case for `aroundLatLng`**: filter the `_geoloc` locations with the `matchedGeoLocation`  value retrieved from the `_rankingInfo` attribute
- **case for `insideBoundingBox`**: filter the `_geoloc` locations from the current map bounding box. It means that it might happen that we have more results on the map rather than in the hits. Because a single hit might have multiple locations that match the current map view.

**Comment by berserkia** [comment]
Any updates or plans to implement this? Or wondering if anyone has successfully forked and implemented a solution.

**Comment by Haroenv** [comment]
There wasn't much requests for this feature, which is why we didn't look into it yet. However the code should be here if you're interested in contributing: 

- expand the hits to be multiple per item here: https://github.com/algolia/instantsearch.js/blob/master/src/connectors/geo-search/connectGeoSearch.ts#L328
- the locations mentioned in OP

@berserkia, alternatively you can also create only one hit per location
